### ISSUE TITLE
GetBookTrail()

### ISSUE THREAD

**Issue by deSagaz** [edushifts/book-voyage#2]
This query will be used to render maps.

**GetBookTrail()**
- Takes `BookInstance.id`
- Returns a matrix with `{holder.first_name, holder.last_name, lat, lon, timestamp, message}` for each holder in chronological order.

*Perhaps a GetBookHandover() can be added as an intermediary function to get each row*

**Comment by deSagaz** [comment]
**Pseudocode by @virtenyatko :**
```def getBookTrail(id):
    for book in BookInstance:
        if book.id == id:
            rightBook = book
    if rightBook in BookInstance:
        holdersList=[]
        for holder in rightBook.holders:
            for location in UserLocation:
                if location.user == holder and location.time == rightBook.holders.time:
                    lon = location.lon
                    lat = location.lat
                    time = location.time
                    message = rightBook.holders.message
            holdersList.append([holder.first_name, holder.last_name,lon,lat,time,message])
        return holdersList
    else:
        return BaseException

**Comment by virtenyatko** [comment]
`def getBookTrail(id):
    try:
        book = BookInstance.objects.get(id=id)
    except BookInstance.DoesNotExist:
        book = -1
    result=[]
    try:
        holdedBook = BookHolding.objects.get(book=book)
        try:
            location = UserLocation.objects.get(user=holdedBook.holder)
            if location.time == holdedBook.time:
                lon = location.lon
                lat = location.lat
                time = location.time
                message = holdedBook.message
                result.append(holdedBook.holder.first_name,holdedBook.holder.last_name,lon,lat,time,message)
        except UserLocation.DoesNotExist:
            pass
    except BookHolding.DoesNotExist:
        pass
    return result`

**Comment by virtenyatko** [comment]
def getBookTrail(id):
    try:
        book = BookInstance.objects.get(id=id)
    except BookInstance.DoesNotExist:
        book = -1
    result=[]
    try:
        holdedBook = BookHolding.objects.get(book=book)
        try:
            location = UserLocation.objects.get(user=holdedBook.holder)
            if location.time == holdedBook.time:
                lon = location.lon
                lat = location.lat
                time = location.time
                message = holdedBook.message
                result.append(holdedBook.holder.first_name,holdedBook.holder.last_name,lon,lat,time,message)
        except UserLocation.DoesNotExist:
            pass
    except BookHolding.DoesNotExist:
        pass
    return result

**Comment by virtenyatko** [comment]
```
def getBookTrail(id):
    try:
        book = BookInstance.objects.get(id=id)
    except BookInstance.DoesNotExist:
        book = -1
    result=[]
    try:
        holdedBook = BookHolding.objects.get(book=book)
        try:
            location = UserLocation.objects.get(user=holdedBook.holder)
            if location.time == holdedBook.time:
                lon = location.lon
                lat = location.lat
                time = location.time
                message = holdedBook.message
                result.append(holdedBook.holder.first_name,holdedBook.holder.last_name,lon,lat,time,message)
        except UserLocation.DoesNotExist:
            pass
    except BookHolding.DoesNotExist:
        pass
    return result
```

**Comment by deSagaz** [comment]
Implemented as of 2bae2da7699e3a636983f0d29a8a4dcdd9c928fa
